sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/ShuffleExchangeExec.scala

```diff
@@ -264,7 +264,11 @@ object ShuffleExchangeExec {
             prefixComparator,
             prefixComputer,
             pageSize,
-            canUseRadixSort)
+            // As we need to compare the binary of UnsafeRow here, we can't make sure whether all
+            // the fields can sort fully with prefix like SortExec. So we disable radix sort here
+            // to avoid getting unstable sort, and result to a correctness bug.
+            // See more details in SPARK-28699.
```

Comment:
how about
```
we are comparing binary here, which does not support radix sort. See more details in SPARK-28699.
```

Reply:
Copy, shorten it in 8bf4cc8.

Reply:
Hi, @cloud-fan . Is this the only one for this topic? Or, do we have more TODO items against old branches (`branch-2.4` and `branch-2.3`)?

Reply:
I have one more: https://github.com/apache/spark/pull/25498 , which will be merged soon, after tests pass.

This one may need more time. We need to collect feedback about whether we should disable "sort before repartition" by default. cc @jiangxb1987